lib/middleware/groups-required.js

```diff
@@ -4,63 +4,89 @@ var helpers = require('../helpers');
 
 /**
  * Assert that a user is a member of one or more groups before allowing the user
- * to continue.  If the user is not logged in, they will be redirected to the
- * login page.  If the user does not meet the group requirements, they will be
+ * to continue. If the user is not logged in, they will be redirected to the
+ * login page. If the user does not meet the group requirements, they will be
  * shown an unauthorized page letting them know they do not have the required
  * permissions.
  *
- * @param {String[]} groups - A list of groups to assert membership in.  Groups
+ * @param {String[]} assertGroups - List of groups to assert membership for. Groups
  *   must be specified by group name.
- * @param {Boolean} [all=true] - Should we assert the user is a member of all groups,
- *   or just one?
+ * @param {Boolean} [all=true] - Whether membership is required in one group or all.
  *
  * @returns {Function} Returns an express middleware which asserts a user's
  *   group membership, and only allows the user to continue if the assertions
  *   are true.
  */
-module.exports = function (groups, all) {
+module.exports = function (assertGroups, all) {
   all = all === false ? false : true;
 
   return function (req, res, next) {
     var config = req.app.get('stormpathConfig');
-    var logger = req.app.get('stormpathLogger');
-    var view = 'unauthorized';
 
-    if (!req.user) {
+    function redirectToLoginPage() {
       var url = config.web.login.uri + '?next=' + encodeURIComponent(req.originalUrl);
       return res.redirect(302, url);
     }
 
-    // If this user must be a member of all groups, we'll ensure that is the
-    // case.
-    var done = groups.length;
-    var safe = false;
+    function isUserInGroups(callback) {
+      var isInGroup = false;
+      var done = assertGroups.length;
 
-    req.user.getGroups(function (err, grps) {
-      if (err) {
-        logger.info('Could not fetch user ' + req.user.email + '\'s groups.');
-        return helpers.render(req, res, view);
-      }
+      req.user.getGroups(function (err, userGroups) {
+        if (err) {
+          return callback(null, false);
```

Comment:
This error should be sent to `logger.error()`, so that the developer can know what went wrong.